test/tests/functional/pbs_job_array.py

```diff
@@ -114,18 +115,68 @@ def test_arraysubjob_survive_restart(self):
                            j.create_subjob_id(j_id, 1), max_attempts=1)
 
         # 6. subjob 2 should be R
-        self.server.expect(JOB, {'job_state': 'R'},
-                           j.create_subjob_id(j_id, 2), max_attempts=1)
+        self.server.expect(JOB, {'job_state': 'R'}, subjid_2, max_attempts=1)
 
         # 7. subjob 3 should be Q
         self.server.expect(JOB, {'job_state': 'Q'},
                            j.create_subjob_id(j_id, 3), max_attempts=1)
 
-    def test_arraysubjob_survive_restart_with_history(self):
+    def test_running_subjob_survive_restart_with_history(self):
         """
-        Test to check if subjobs of an array job survive a pbs_server restart
-        when history is enabled
+        Test to check if a running subjob of an array job survive a
+        pbs_server restart when history is enabled
         """
         attr = {'job_history_enable': 'true'}
         self.server.manager(MGR_CMD_SET, SERVER, attr)
-        self.test_arraysubjob_survive_restart()
+        self.test_running_subjob_survive_restart()
+
+    def test_suspended_subjob_survive_restart(self):
+        """
+        Test to check if a suspended subjob of an array job survive a
+        pbs_server restart
+        """
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        j = Job(TEST_USER, attrs={
+            ATTR_J: '1-3', 'Resource_List.select': 'ncpus=1'})
+
+        j.set_sleep_time(10)
+
+        j_id = self.server.submit(j)
+        subjid_2 = j.create_subjob_id(j_id, 2)
+
+        # 1. check job array has begun
+        self.server.expect(JOB, {'job_state': 'B'}, j_id)
+
+        # 2. wait till subjob 2 starts running
```

Comment:
Update the comment with subjob_2

Reply:
Fixed in rc 33 https://github.com/PBSPro/pbspro/pull/590/commits/3da33d42343d89615cf55f6531ed5362ebcdd287